### PR TITLE
Use Jest for tests

### PR DESCRIPTION
**Pull request by turt2live**
Fixes https://github.com/vector-im/riot-web/issues/11291

**See commits for a bit of a story to how we got here.**

There's a couple things of note that aren't recorded in the diff very well:
* We bring in `canvas` for the COLR checks to pass (and upgrade our jsdom environment to pull it in correctly).
* We use a fake indexeddb so we can do indexeddb things.
* We have to map some modules to things to fix i18n, the number of copies of react we're running, etc
* The tests take about 166 seconds to run on my machine (i9-9900k, many RAM) - they surely take longer on CI.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -116,28 +116,20 @@
...
     "eslint-plugin-react": "^7.11.1",
     "eslint-plugin-react-hooks": "^2.2.0",
-    "expect": "^1.16.0",
     "extract-text-webpack-plugin": "^4.0.0-beta.0",
+    "fake-indexeddb": "^3.0.0",
     "file-loader": "^5.0.2",
     "fs-extra": "^0.30.0",
     "html-webpack-plugin": "^3.2.0",
+    "jest": "^24.9.0",
+    "jest-environment-jsdom-thirteen": "^1.0.1",
```
Comment (t3chguy) [inline package.json]: any reason we're using JSDOM13 when the https://www.npmjs.com/package/jest-environment-jsdom-thirteen page basically tells you to use 14?

File: package.json [hunk 2]
```diff
@@ -198,5 +190,26 @@
...
+    ],
+    "setupFilesAfterEnv": [
+      "<rootDir>/node_modules/matrix-react-sdk/test/setupTests.js"
+    ],
+    "moduleNameMapper": {
+      "\\.(gif|png|svg|ttf|woff2)$": "<rootDir>/node_modules/matrix-react-sdk/__mocks__/imageMock.js",
+      "\\$webapp/i18n/languages.json": "<rootDir>/node_modules/matrix-react-sdk/__mocks__/languages.json",
+      "^browser-request$": "<rootDir>/node_modules/matrix-react-sdk/__mocks__/browser-request.js",
+      "^react$": "<rootDir>/node_modules/react",
+      "^react-dom$": "<rootDir>/node_modules/react-dom"
```
Comment (t3chguy) [inline package.json]: what the heck do these resolve to without this?

File: test/app-tests/loading-test.js [hunk 3]
```diff
@@ -94,13 +81,14 @@ describe('loading:', function() {
         expect(MatrixClientPeg.get()).toBe(null);
 
         // chrome seems to take *ages* to delete the indexeddbs.
-        this.timeout(10000);
+        await sleep(10000);
```
Comment (t3chguy) [inline test/app-tests/loading-test.js]: Tests aren't running in chrome anymore, is this still needed?

File: package.json (hunk 1 above)
Comment (turt2live) [reply package.json]: The Google told me to. Will test with fourteen and see what happens (probably nothing breaking)

File: package.json (hunk 2 above)
Comment (turt2live) [reply package.json]: well, that's the fun thing: when the react-sdk references React[DOM] itself, such as the case when you have it `yarn link`ed, it puts two React[DOM]s in your bundle. This tells it to use one.

File: test/app-tests/loading-test.js (hunk 3 above)
Comment (turt2live) [reply test/app-tests/loading-test.js]: nope, apparently not. Have removed.

File: package.json (hunk 1 above)
Comment (turt2live) [reply package.json]: 14 looks fine

**Pull request merged**